### DISCUSSION TITLE
Building espeak (fresh MXlinux install) Target Android 14.

### DISCUSSION THREAD

**Discussion by BobSquarePants** [espeak-ng/espeak-ng#1921] [Q&A] [answered]
Hi,

Since there is no yet reaction on my [previous post](https://github.com/orgs/espeak-ng/discussions/1917)

I've decided to install a MX Linux from scratch and build the `espeak-ng.apk` following [this guide.](https://github.com/espeak-ng/espeak-ng/blob/master/docs/building.md#android)

Here what installed (ordered)

.. xx (are the dependency)

1. MX linux
2. Android studio 2023.3.1.19 (standard install)
3. NDK (Side by side) within Android studio
4. (all the following trough synaptic)
5. openjdk-17-jdk
   .. openjdk-17-jdk-headless
6. automake
        .. autoconf
        .. autotools-dev
        .. m4
7. gradle
.. ant (version 1.10.13-1)
.. ant-optional (version 1.10.13-1)
.. antlr (version 2.7.7+dfsg-12)
.. bnd (version 5.0.1-3)
.. gradle (version 4.4.1-18)
.. groovy (version 2.4.21-8)
.. ivy (version 2.5.1-2)
.. java-wrappers (version 0.4)
.. junit4 (version 4.13.2-3)
.. libantlr-java (version 2.7.7+dfsg-12)
.. libaopalliance-java (version 20070526-7)
.. libapache-pom-java (version 29-2)
.. libasm-java (version 9.4-1)
.. libatinject-jsr330-api-java (version 1.0+ds1-5)
.. libbcel-java (version 6.5.0-2)
.. libbcpg-java (version 1.72-2)
.. libbcprov-java (version 1.72-2)
.. libbsf-java (version 1:2.4.0-8)
.. libbsh-java (version 2.0b4-20)
.. libcdi-api-java (version 1.2-3)
.. libcommons-cli-java (version 1.5.0-1)
.. libcommons-codec-java (version 1.15-1)
.. libcommons-collections3-java (version 3.2.2-2)
.. libcommons-compress-java (version 1.22-1)
.. libcommons-io-java (version 2.11.0-2)
.. libcommons-lang-java (version 2.6-10)
.. libcommons-lang3-java (version 3.12.0-2)
.. libcommons-logging-java (version 1.2-3)
.. libcommons-parent-java (version 56-1)
.. libdd-plist-java (version 1.20-1.1)
.. libdom4j-java (version 2.1.3-2)
.. libeclipse-jdt-annotation-java (version 2.2.700+eclipse4.26-2)
.. liberror-prone-java (version 2.18.0-1)
.. libfelix-framework-java (version 4.6.1-2.1)
.. libfelix-gogo-runtime-java (version 0.16.2-1.1)
.. libfelix-resolver-java (version 1.16.0-1)
.. libfindbugs-java (version 3.1.0\~preview2-3)
.. libgeronimo-annotation-1.3-spec-java (version 1.3-1)
.. libgeronimo-interceptor-3.0-spec-java (version 1.0.1-4)
.. libgoogle-gson-java (version 2.10-1)
.. libgradle-core-java (version 4.4.1-18)
.. libgradle-plugins-java (version 4.4.1-18)
.. libguava-java (version 31.1-1)
.. libguice-java (version 4.2.3-2)
.. libhamcrest-java (version 2.2-1)
.. libhawtjni-runtime-java (version 1.18-1)
.. libhttpclient-java (version 4.5.14-1)
.. libhttpcore-java (version 4.4.16-1)
.. libjansi-java (version 2.4.0-2)
.. libjansi-native-java (version 1.8-1)
.. libjansi1-java (version 1.18-3)
.. libjarjar-java (version 1.4+svn142-12)
.. libjatl-java (version 0.2.3-1.1)
.. libjavaewah-java (version 1.1.7-1)
.. libjaxen-java (version 1.1.6-4)
.. libjcifs-java (version 1.3.19-2)
.. libjcip-annotations-java (version 20060626-6)
.. libjcommander-java (version 1.71-4)
.. libjetty9-java (version 9.4.50-4+deb12u3)
.. libjformatstring-java (version 0.10\~20131207-2.1)
.. libjgit-java (version 4.11.9-2)
.. libjline2-java (version 2.14.6-5)
.. libjna-java (version 5.13.0-2)
.. libjna-jni (version 5.13.0-2)
.. libjs-jquery (version 3.6.1+dfsg+\~3.5.14-1)
.. libjsch-java (version 0.1.55-1)
.. libjsoup-java (version 1.15.3-1)
.. libjsr305-java (version 0.1\~+svn49-11)
.. libjzlib-java (version 1.1.3-2)
.. libkryo-java (version 2.20-7)
.. liblogback-java (version 1:1.2.11-3)
.. libmaven-parent-java (version 35-1)
.. libmaven-resolver-java (version 1.6.3-1)
.. libmaven-shared-utils-java (version 3.3.4-1)
.. libmaven3-core-java (version 3.8.7-1)
.. libminlog-java (version 1.3.0-1.1)
.. libnative-platform-java (version 0.14-5)
.. libnative-platform-jni (version 0.14-5)
.. libnekohtml-java (version 1.9.22.noko2-0.1)
.. libobjenesis-java (version 3.3-3)
.. libosgi-annotation-java (version 8.1.0-1)
.. libosgi-compendium-java (version 7.0.0-1)
.. libosgi-core-java (version 8.0.0-2)
.. libplexus-cipher-java (version 2.0-1)
.. libplexus-classworlds-java (version 2.7.0-1)
.. libplexus-component-annotations-java (version 2.1.1-1)
.. libplexus-container-default-java (version 2.1.1-1)
.. libplexus-interpolation-java (version 1.26-1)
.. libplexus-sec-dispatcher-java (version 2.0-3)
.. libplexus-utils2-java (version 3.4.2-1)
.. libpolyglot-maven-java (version 0.8\~tobrien+git20120905-10)
.. libqdox-java (version 1.12.1-3)
.. libreflectasm-java (version 1.11.9+dfsg-4)
.. librhino-java (version 1.7.14-2.1)
.. libsimple-http-java (version 4.1.21-1.1)
.. libsisu-inject-java (version 0.3.4-2)
.. libsisu-plexus-java (version 0.3.4-3)
.. libslf4j-java (version 1.7.32-1)
.. libwagon-file-java (version 3.5.3-1)
.. libwagon-http-java (version 3.5.3-1)
.. libwagon-provider-api-java (version 3.5.3-1)
.. libxbean-reflect-java (version 4.5-8)
.. libxerces2-java (version 2.12.2-1)
.. libxml-commons-external-java (version 1.4.01-5)
.. libxml-commons-resolver1.1-java (version 1.2-11)
.. libxpp3-java (version 1.1.4c-3)
.. libxstream-java (version 1.4.20-1)
.. libxz-java (version 1.9-1)
.. libyaml-snake-java (version 1.33-2)
.. testng (version 6.9.12-4)

After this I `cd` into the unpacked [sources](https://github.com/espeak-ng/espeak-ng/archive/refs/tags/1.51.zip)

did
```bash
export ANDROID_HOME=/home/user/Android/Sdk
```

but
```bash
./autogen.sh
```
returned
> ./autogen.sh: 16: --copy: not found

Any ideas... ?

**Comment by BobSquarePants** [comment]
@RetroBunn , how did you manage to build it ? https://github.com/espeak-ng/espeak-ng/issues/1862#issuecomment-2081310179

**Comment by BobSquarePants** [comment]
In order to solve 

> ./autogen.sh: 16: --copy: not found

You need to install `libtool` I've also installed `libtool-bin` without knowing if it's required or not..

Let see if I can finish that built...

**Reply by BobSquarePants** [reply]
Of course I get a new error

```bash
make apk-release
```
give
> Makefile:2883: warning: ignoring prerequisites on suffix rule definition
cd android && ndk-build
/bin/bash: line 1: ndk-build: command not found
make: *** [Makefile:3081: jni] Error 127

I tried to modify my `$PATH` with fews `export PATH=$PATH:/home/aUser/Android/Sdk/ndk:/home/aUser/Android/Sdk/ndk/27.0.11718014/ndk-build`

But I still get the same error :/

**Comment by alex19EP** [comment, marked as answer]
use latest master. and just build with gradel no autotools required

**Reply by BobSquarePants** [reply]
Thanks @alex19EP ,

I've downloaded [gradle v8.8](https://gradle.org/releases/)

I've done
```bash
./configure --with-gradle=/home/someone/Desktop/gradle88
```

but I don't see (like [mentioned here](https://github.com/espeak-ng/espeak-ng/blob/master/docs/building.md#building-with-gradle) )

> ...
> gradle (Android):              gradle
> ...


in the output :/

I got
>gradle (Android):  


any ideas... ?

**Reply by BobSquarePants** [reply]
never mind I've put `/home/someone/Desktop/gradle88` in `$PATH` and it's now displaying.
